### PR TITLE
Cpp master issue 994

### PR DESCRIPTION
**Pull request by sukidog**

### CODE REVIEW

**Comment by redboltz** [comment]
Thank you for quick post.
I cross in the post https://github.com/msgpack/msgpack-c/issues/994#issuecomment-963237955
Please see it.

**Comment by sukidog** [comment]
> Thank you for quick post. I cross in the post [#994 (comment)](https://github.com/msgpack/msgpack-c/issues/994#issuecomment-963237955) Please see it.

Solution adjusted as per your feedback. See new commit

File: include/msgpack/v1/object.hpp [hunk 1]
```diff
@@ -466,12 +466,13 @@ struct object_stringize_visitor {
...
-    bool visit_bin(const char* v, uint32_t size) {
-        (m_os << '"').write(v, static_cast<std::streamsize>(size)) << '"';
+    bool visit_bin(const char* /*v*/, uint32_t size) {
+        m_os << "\"BIN(size:" << size << ")\"";
         return true;
     }
-    bool visit_ext(const char* /*v*/, uint32_t /*size*/) {
-        m_os << "EXT";
+    bool visit_ext(const char* v, uint32_t size) {
+        int type = ((size > 0) && (v[0] >= 0)) ? v[0] : -1;
```
Comment (redboltz) [inline include/msgpack/v1/object.hpp:474]: Maybe my comment is misleading. If the value of type is `-2` the output should be `-2`. I mean print as **singed**.

File: include/msgpack/v1/object.hpp [hunk 2]
```diff
@@ -466,12 +466,13 @@ struct object_stringize_visitor {
...
+        m_os << "\"EXT(type:" << type << ",size:" << size << ")\"";
```
Comment (redboltz) [inline include/msgpack/v1/object.hpp:475]: See https://github.com/msgpack/msgpack/blob/master/spec.md#ext-format-family
The parameter `size` is size of type + data.
I think that users expect size of data. So the printed value should size-1.

Comment (redboltz) [reply include/msgpack/v1/object.hpp:475]: So the code should be

```cpp
    bool visit_ext(const char* v, uint32_t size) {
        if (size == 0) {
            m_os << "\"EXT(size:0)\"";
        }
        else {
            m_os << "\"EXT(type:" << static_cast<int>(v[0]) << ",size:" << size - 1 << ")\"";
        }
```

Comment (sukidog) [reply include/msgpack/v1/object.hpp:475]: Got it. Thanks. Will upload a change tomorrow morning.

Comment (redboltz) [reply include/msgpack/v1/object.hpp:475]: NOTE: size==0 on EXT is error. But I choose printing only size. (avoid invalid v[0] access.

**Comment by sukidog** [comment]
I uploaded the EXT implementation you asked for.

**Comment by redboltz** [comment]
Looks good to me.
Thank you for the contribution!

**Comment by sukidog** [comment]
Thanks for accepting it so quickly!

**Pull request merged**